### PR TITLE
Add signIn module and social, logged in and two factor components

### PR DESCRIPTION
**Pull request by DanielPalafox**
https://trello.com/c/yDZoD0Sq/6541-signin-user-name-and-password-13

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[Signin - user name and password (13)](https://trello.com/c/yDZoD0Sq/6541-signin-user-name-and-password-13)

**Comment by DanielPalafox** [comment]
[Signin - user name and password (13)](https://trello.com/c/yDZoD0Sq/6541-signin-user-name-and-password-13)

**Comment by DanielPalafox** [comment]
[Migrate 2FA to the new UI](https://trello.com/c/h2ZCS40c/6665-migrate-2fa-to-the-new-ui)

File: src/app/sign-in/sign-in.module.ts [hunk 1]
```diff
@@ -2,11 +2,46 @@ import { NgModule } from '@angular/core'
 import { CommonModule } from '@angular/common'
 import { SignInComponent } from './pages/sign-in/sign-in.component'
 import { SignInRoutingModule } from './sign-in-routing.module'
-import { MatDialogModule } from '@angular/material'
+import {
+  MatButtonModule,
+  MatCardModule,
+  MatChipsModule,
```
Comment (leomendoza123) [inline src/app/sign-in/sign-in.module.ts:8]: I believe the chip and dialog modules are not required on this module

File: src/app/sign-in/pages/sign-in/sign-in.component.ts [hunk 2]
```diff
@@ -1,13 +1,132 @@
...
+            environment.BASE_URL + info.userInfo.REAL_USER_ORCID
+        })
+      }
+    })
+  }
+
+  usernameFormControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(
+      '(\\d{4}-){3}\\d{3}[\\dX]|^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,4}$'
```
[the remark this answers is not in the document]
Comment (amontenegro) [reply src/app/sign-in/pages/sign-in/sign-in.component.ts:56]: I think we should use option B

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -30,11 +30,16 @@ const routes: Routes = [
...
+  {
+    path: ApplicationRoutes.login,
+    loadChildren: () =>
+      import('./sign-in/sign-in.module').then(m => m.SignInModule),
+  },
+  {
+    path: ApplicationRoutes.signin,
+    loadChildren: () =>
+      import('./sign-in/sign-in.module').then(m => m.SignInModule),
+  },
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/app-routing.module.ts:41]: Done!

File: src/app/sign-in/components/errors/deactivated/deactivated.component.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
+p {
+  font-size: 14px;
+  font-weight: normal;
+  line-height: 1.43;
+}
+
+a {
+  font-size: 14px;
```
[the remark this answers is not in the document]
Comment (DanielPalafox) [reply src/app/sign-in/components/errors/deactivated/deactivated.component.scss]: Ok

**Comment by DanielPalafox** [comment]
@leomendoza123 I have implemented all the changes you suggested

File: src/app/sign-in/components/two-factor/two-factor.component.scss [hunk 5]
```diff
@@ -1,7 +1,20 @@
+.mat-button-font {
+  letter-spacing: 1.25px;
```
Comment (leomendoza123) [inline src/app/sign-in/components/two-factor/two-factor.component.scss:2]: I believe this is letter-spacing is not required

Comment (DanielPalafox) [reply src/app/sign-in/components/two-factor/two-factor.component.scss:2]: ok I will remove it

File: src/app/sign-in/pages/sign-in/sign-in.component.ts [hunk 6]
```diff
@@ -1,13 +1,139 @@
...
+      })
+  }
+
+  usernameFormControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(
+      '(\\d{4}-){3}\\d{3}[\\dX]' +
+        '|^.*\\.([a-zA-Z\\-])([a-zA-Z\\-]{0,61})([a-zA-Z\\-])$'
+    ),
+  ])
```
Comment (leomendoza123) [inline src/app/sign-in/pages/sign-in/sign-in.component.ts:65]: @DanielPalafox I think I explain wrong with this one https://github.com/ORCID/orcid-angular/pull/148#discussion_r408546091

We do need to create a new custom validator, to validate "orcidOrEmail" 

Something like

```
  static orcidOrEmail(pattern: string | RegExp): ValidatorFn {
    return (control: AbstractControl) => {

      If (Validators.email(control) || Validators.pattern(TLD_REGEXP)(control) && Validators.pattern(ORCID_REGEXP)(control) ){
    return {'invalidUserName': true}
    }
    return null
    }
  }
```

After that use that validator on the usernameFormControl  like

```
usernameFormControl = new FormControl('', [
    Validators.required,
    this.isEmailOrPassword()
```

The recent change to this code only uses the TLD_REGEXP for emails but that only works well in combination with the Angular email validator. So this custom validator should solve that

Note:  please import constants from the constants.ts file when those are required.

**Pull request merged**